test-harness

```diff
@@ -57,15 +57,24 @@ def check_expected(name, item, got, expected, errors):
     else:
         return True
 
-def evaluate(name, code, stdout='', stderr='', exit = '0', flags='-e', env = None, filemode='', args='', ignore = None):
+def evaluate(name, code, config_file, stdout='', stderr='', exit = '0', env = None, filemode='', args='', ignore = None):
+    arg_array = str.split(args)
+    if config_file != None:
+        for arg in arg_array:
+            if arg.startswith("--config"):
+                print(
+                    "Test \"%s\" comes with an args entry to specify the config file," % name +
+                    " but a config file was also passed as a command-line argument to the test harness" )
+                exit(1)
+
+        arg_array += ["--config=" + config_file]
+
+
     if filemode.startswith('true') :
-        proc = Popen([links, code], stdout=PIPE, stderr=PIPE, env=env)
-    elif filemode.startswith('args'):
-        arg_array = str.split(args, ' ')
-        popen_list = [links] + arg_array + [code]
-        proc = Popen(popen_list, stdout=PIPE, stderr=PIPE, env=env)
+        proc = Popen([links] + arg_array + [code], stdout=PIPE, stderr=PIPE, env=env)
```

Comment:
I'd avoid the repetition with something like (I don't know Python, so syntax may be wrong):

```Python
if filemode.startswith('true') :
    arg_array = arg_array + "-e"

proc = Popen([links] + arg_array + [code], stdout=PIPE, stderr=PIPE, env=env)
```

Reply:
Good point. Done!